src/reference/asciidoc/smb.adoc

```diff
@@ -0,0 +1,109 @@
+compile "org.springframework.integration:spring-integration-smb:{project-version}"
+----
+====
+
+=== Overview
+
+The Java CIFS Client Library has been chosen as a Java implementation for the CIFS/SMB networking protocol.
+Its `SmbFile` abstraction is simply wrapped to the Spring Integration "Remote File" foundations like `SmbSession`, `SmbRemoteFileTemplate`, etc.
+
+The SMB Channel Adapters and support classes implementations are fully similar to existing components for (S)FTP or AWS S3 protocols.
+So, if you familiar with those components it is pretty straightforward to use.
+
+[[smb-session-factory]]
+=== SMB Session Factory
+
+Before configuring the SMB adapter, you must configure a SMB session factory.
+You can configure the SMB session factory with a regular bean definition, as the following examples show:
+
+==== Setting the SMB protocol with Min/Max versions
+For supporting a minimum version of SMB 2.1 and a maximum version of the SMB 3.1.1
+
+[source,java]
+----
+@Bean
+public SmbSessionFactory smbSessionFactory() {
+    SmbSessionFactory smbSession = new SmbSessionFactory();
+    smbSession.setHost("myHost");
+    smbSession.setPort(445);
+    smbSession.setDomain("myDomain");
+    smbSession.setUsername("myUser");
+    smbSession.setPassword("myPassword");
+    smbSession.setShareAndDir("myShareAndDir");
+    smbSession.setSmbMinVersion(DialectVersion.SMB210);
+    smbSession.setSmbMaxVersion(DialectVersion.SMB311);
+    return smbSession;
+}
+----
+
+==== Initializing SmbSessionFactory with a custom implementation of the jcifs.CIFSContext interface
+
+NOTE: Setting of the SMB protocol Min/Max versions must be done in your implementation of jcifs.CIFSContext
+
+[source,java]
+----
+@Bean
+public SmbSessionFactory smbSessionFactory() {
+    SmbSessionFactory smbSession = new SmbSessionFactory(new MyCIFSContext());
+    smbSession.setHost("myHost");
+    smbSession.setPort(445);
+    smbSession.setDomain("myDomain");
+    smbSession.setUsername("myUser");
+    smbSession.setPassword("myPassword");
+    smbSession.setShareAndDir("myShareAndDir");
+    return smbSession;
+}
+----
+
+[[smb-inbound]]
+=== SMB Inbound Channel Adapter
+
+To download SMB files locally the `SmbInboundFileSynchronizingMessageSource` is provided.
+It is simple extension of the `AbstractInboundFileSynchronizingMessageSource` which  requires `SmbInboundFileSynchronizer` injection.
+For filtering remote files you still can use any existing `FileListFilter` implementations, but particular `SmbRegexPatternFileListFilter` 
+and `SmbSimplePatternFileListFilter` are provided.
+
+For XML configuration the `<int-smb:inbound-channel-adapter>` component is provided.
+
+[[smb-outbound]]
+=== SMB Outbound Channel Adapter
+
+For writing files to a SMB share, and for XML `<int-smb:outbound-channel-adapter>` component we use the `SmbMessageHandler`.
+In case of Java configuration a `SmbMessageHandler` should be supplied with the `SmbSessionFactory` (or `SmbRemoteFileTemplate`).
```

Comment:
I think we have reinstate then a ctor in the `SmbMessageHandler` based on the `SmbRemoteFileTemplate`, like we have in other `FileTransferringMessageHandler` impls.

Reply:
Agreed... do you want to take this on?